src/ripple/protocol/impl/STObject.cpp

```diff
@@ -61,10 +61,19 @@ STObject::STObject(SerialIter& sit, SField const& name, int depth) noexcept(
 }
 
 STObject
-STObject::makeInnerObject(SField const& name, Rules const& rules)
+STObject::makeInnerObject(SField const& name)
 {
     STObject obj{name};
-    if (rules.enabled(fixInnerObjTemplate))
+
+    // The if is complicated because inner object templates were added in
+    // two phases:
+    //  1. If there are no available Rules, then always apply the template.
```

Comment:
Why should the template be applied if there is no `Rules`?

Reply:
@gregtatcam, great question.

The way `getCurrentTransactionRules()` works is:

1. If we're currently inside a transaction then the `Rules` are set and available.
2. If this code is ever called when we're not inside a transaction, then the optional is unseated.

This means that if the optional is unseated then we're not inside a transaction, and any changes we make will not be transaction breaking.  We have two choices we could make at this point:

1. If we're not inside a transaction, stick with the old behavior.
2. If we're not inside a transaction apply the new behavior.

In my opinion the new behavior (apply the template) is strictly safer than the old behavior (don't apply the template).  So I'm choosing to bias the change to honor safety.

Thoughts?

Reply:
Thanks for the clarification. Applying the new behavior sounds right to me. One argument to use the new behavior is when a unit-test doesn't call a transaction but does call a function with this amendment, for instance some utility. We want to make sure that the unit-test fails in this case if there is a difference in before/after.